### PR TITLE
Adding code for Longest Substring Without Repeating Characters

### PR DESCRIPTION
**Pull request by anilhariparamhans**
https://leetcode.com/problems/longest-substring-without-repeating-characters/

### CODE REVIEW

File: LeetCode/vAnil/Practice/LongestSubstrWithoutRepeatingChars/LongestSubstrWithoutRepeatingChars.cpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+				maxSeqLen = curSeqCount;
+
+			seqMap.clear();
+			curSeqCount = 1;
+			seqMap.insert(x);
+
+		}
+	}
+
+	return maxSeqLen;
```
Comment (dimplenaresh1) [inline LeetCode/vAnil/Practice/LongestSubstrWithoutRepeatingChars/LongestSubstrWithoutRepeatingChars.cpp]: Greater of maxSeqLen and CurrSeqCount should be returned.

**Review by dimplenaresh1 — CHANGES_REQUESTED** [review]

File: LeetCode/vAnil/Practice/LongestSubstrWithoutRepeatingChars/LongestSubstrWithoutRepeatingChars.cpp (hunk 1 above)
Comment (anilhariparamhans) [reply LeetCode/vAnil/Practice/LongestSubstrWithoutRepeatingChars/LongestSubstrWithoutRepeatingChars.cpp]: Good catch !

I incorporated this review comment.

File: LeetCode/vAnil/Practice/PracticeQuestions/ReverseWordInString.cpp [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+		for (size_t j = start; j <= end ; j++)
+		{
+			char temp = str[j];
+			str[j] = str[end];
+			str[end] = temp;
+			end--;
+		}
+	}
+
+	return str;
```
Comment (dimplenaresh1) [inline LeetCode/vAnil/Practice/PracticeQuestions/ReverseWordInString.cpp:86]: getting SEGV error when submitting in Leetcode.

**Review by dimplenaresh1 — CHANGES_REQUESTED** [review]

File: LeetCode/vAnil/Practice/PracticeQuestions/ReverseWordInString.cpp [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	for (size_t i = 0; i < len; i++)
+	{
+		if (str[i] == ' ')
+		{
+			count++;
+		}
+		else
+			break;
+	}
+	str.erase(0, count);
```
Comment (dimplenaresh1) [inline LeetCode/vAnil/Practice/PracticeQuestions/ReverseWordInString.cpp:18]: out of range subscript error if str is ""

File: LeetCode/vAnil/Practice/PracticeQuestions/ReverseWordInString.cpp (hunk 2 above)
Comment (anilhariparamhans) [reply LeetCode/vAnil/Practice/PracticeQuestions/ReverseWordInString.cpp:86]: Fixed this.

File: LeetCode/vAnil/Practice/PracticeQuestions/ReverseWordInString.cpp (hunk 3 above)
Comment (anilhariparamhans) [reply LeetCode/vAnil/Practice/PracticeQuestions/ReverseWordInString.cpp:18]: Fixed it.

**Comment by anilhariparamhans** [comment]
@dimplenaresh1 fixed the issues
and got this result on Leetcode

Runtime: 8 ms, faster than 99.68% of C++ online submissions for Reverse Words in a String.
Memory Usage: 9.6 MB, less than 41.55% of C++ online submissions for Reverse Words in a String.

**Review by dimplenaresh1 — APPROVED** [review]

**Pull request merged**